core/src/default_components.rs

```diff
@@ -128,6 +133,18 @@ where
         self.dispatcher.wait_ended();
         self.deadletter_box.wait_ended();
     }
+
+    fn connect_network_status_port(&self, required: &mut RequiredPort<NetworkStatusPort>) -> () {
+        self.dispatcher.on_definition(|d| {
+            let any_port = d
+                .get_provided_port_as_any(TypeId::of::<NetworkStatusPort>())
```

Comment:
Can we not guarantee this statically via extending `Dispatcher` or adding another trait requirement?